### PR TITLE
Introduce --display-name, change generator loop (debug.py)

### PR DESCRIPTION
**Pull request by nikostsagk**
Hello to everybody! This is my first pull request ever so try to show some mercy please :P.
This repo is great, so I will try sharing some modifications I am making while working on my project and brought me some  convenience.

To start with, I propose a --display-name parser where displays on the bottom left corner of the image the name of the image. Looks that it works well when the image is in default margins (800x1300), but needs fontScaling for smaller dimensions (e.g. for image 200x200). 

The "for" loop was a bit buggy after key bindings were introduced, so I changed it to "while True".
Also, introduced restrictions in variable "i" so it can take values only between [0, generator.size()].

P.S: sorry about the keybindings change, it was due to different OS.

I am looking forward for your insights,
Cheers

### CODE REVIEW

File: keras_retinanet/bin/debug.py [hunk 1]
```diff
@@ -154,6 +155,7 @@ def csv_list(string):
     parser.add_argument('-l', '--loop', help='Loop forever, even if the dataset is exhausted.', action='store_true')
     parser.add_argument('--no-resize', help='Disable image resizing.', dest='resize', action='store_false')
     parser.add_argument('--anchors', help='Show positive anchors on the image.', action='store_true')
+    parser.add_argument('--display-name', help='Display image name on the bottom left corner.', dest='name', action='store_true')
```
Comment (hgaiser) [inline keras_retinanet/bin/debug.py]: I don't think there's a need for the `dest='name'` part, having the argument as `dest_name` will be fine.

File: keras_retinanet/bin/debug.py [hunk 2]
```diff
@@ -200,19 +203,30 @@ def run(generator, args, anchor_params):
...
+
+        ## note that the right and left keybindings are probably different for windows
+        ## press right for next image and left for previous (linux)
+        ## macOS: previous = 'n' | next = 'm'
+
+        if key == 109:
             i += 1
-        # press left for previous image
-        if key == 81:
+            if i == (generator.size() - 1): # Terminate after last image.
```
Comment (hgaiser) [inline keras_retinanet/bin/debug.py]: Hmm good catch. Could you change it to the following:

- Remove the `--loop` argument.
- Change the right action to `i = (i + 1) % generator.size()`
- Change the left action to `if i < 0: i = generator.size() - 1`

That way we have looping with the left / right key.

File: keras_retinanet/bin/debug.py [hunk 3]
```diff
@@ -171,7 +173,8 @@ def run(generator, args, anchor_params):
         args: parseargs args object.
     """
     # display images, one at a time
-    for i in range(generator.size()):
+    i=0
```
Comment (hgaiser) [inline keras_retinanet/bin/debug.py]: An equality sign needs spaces around it :)

File: keras_retinanet/bin/debug.py [hunk 4]
```diff
@@ -200,19 +203,30 @@ def run(generator, args, anchor_params):
...
+            if args.name:	
+                draw_caption(image, [0, image.shape[0]], ntpath.basename(generator.image_path(i)))
+                
         cv2.imshow('Image', image)
         key = cv2.waitKey()
-        # note that the right and left keybindings are probably different for windows
-        # press right for next image
-        if key == 83:
+
+        ## note that the right and left keybindings are probably different for windows
```
Comment (hgaiser) [inline keras_retinanet/bin/debug.py]: Double #

File: keras_retinanet/bin/debug.py [hunk 5]
```diff
@@ -200,19 +203,30 @@ def run(generator, args, anchor_params):
...
-        # press left for previous image
-        if key == 81:
+            if i == (generator.size() - 1): # Terminate after last image.
+                return False
+        if key == 110:
             i -= 1
-        # press q to quit
+            if i < 0:                # Don't let i be less than "0".
+                i = 0
+        ## press q to quit
```
Comment (hgaiser) [inline keras_retinanet/bin/debug.py]: Double #

File: keras_retinanet/bin/debug.py [hunk 6]
```diff
@@ -200,19 +203,30 @@ def run(generator, args, anchor_params):
...
         if key == ord('q'):
```
Comment (hgaiser) [inline keras_retinanet/bin/debug.py]: Not really related to this PR, but could you also add the Escape key here?

File: keras_retinanet/bin/debug.py [hunk 7]
```diff
@@ -17,6 +17,7 @@
 """
 
 import argparse
+import ntpath
```
Comment (hgaiser) [inline keras_retinanet/bin/debug.py]: Prefer to use `os.path`, should be cross platform too.

File: keras_retinanet/bin/debug.py (hunk 1 above)
Comment (nikostsagk) [reply keras_retinanet/bin/debug.py]: hmm not sure if i understood, like this: 
```'--display-name', help='Display image name on the bottom left corner.', action='store_true'```

Comment (hgaiser) [reply keras_retinanet/bin/debug.py]: Yeah, so without the `dest`

File: keras_retinanet/bin/debug.py (hunk 2 above)
Comment (nikostsagk) [reply keras_retinanet/bin/debug.py]: nice one!

File: keras_retinanet/bin/debug.py (hunk 7 above)
Comment (nikostsagk) [reply keras_retinanet/bin/debug.py]: that what i was afraid but indeed, it is cross platform

**Comment by nikostsagk** [comment]
Is it better now?

**Comment by hgaiser** [comment]
Yeah, better :)

there's some linting errors to fix though:
https://travis-ci.org/fizyr/keras-retinanet/jobs/557249482

**Comment by hgaiser** [comment]
Thanks again!

**Pull request merged**